Example/Kommunicate/LoginViewController.swift

```diff
@@ -54,13 +62,51 @@ class LoginViewController: UIViewController {
     }
 
     @IBAction func loginAsVisitor(_ sender: Any) {
+        resignFields()
         let applicationId = AppDelegate.appId
         setupApplicationKey(applicationId)
 
         let kmUser = userWithUserId(Kommunicate.randomId(), andApplicationId: applicationId)
         registerUser(kmUser)
     }
 
+    @objc func keyboardWillHide(notification: NSNotification) {
+
+        guard isKeyboardVisible,
+            let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as? TimeInterval
+            else { return }
+        isKeyboardVisible = false
+        UIView.animate(withDuration: duration, delay: 0, options: [.beginFromCurrentState], animations: {
+            self.scrollView.contentInset = self.originalContentInset
+            self.scrollView.scrollIndicatorInsets = self.originalContentInset
+            self.scrollView.contentOffset = CGPoint(x: 0, y: 0)
+        })
+    }
+
+    @objc func keyboardWillShow(notification: NSNotification) {
+
+        guard let frame = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect,
+            let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as? TimeInterval else { return }
+            var inset = scrollView.contentInset
+            originalContentInset = inset
+
+            let converted = self.view.convert(frame, from: nil)
```

Comment:
These variable names are not much explanatory, Can we add some comments like what is happening here and also modify the variable names?